src/plugins/data/README.mdx

```diff
@@ -56,7 +56,73 @@ Coming soon.
 
 ## Index Patterns
 
-Coming soon.
+The Index Patterns API provides a consistent method of structuring and formatting documents
+and field lists across the various Kibana apps. Its typically used in conjunction with
+SearchSource for composing queries.
+
+### Index Patterns API
+
+- Get list of index patterns
+- Get default index pattern and examine fields
+- Get index pattern by id
+- Find index pattern by title
+- Create index pattern
+- Create index pattern and save it
+- Modify index pattern and save it
+- Delete index pattern
+
+#### Get list of index patterns
```

Comment:
Could it be that the title actually doesn't really describe what the API does accurately?
Because it doesn't really fetch the index patterns, but only the id and title, no?